### DISCUSSION TITLE
Add enterprise local database (such as MySQL) as a data source in the RAG function.

### DISCUSSION THREAD

**Discussion by ryanwong2023** [langgenius/dify#3549] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

In enterprise-level application scenarios, in addition to various types of unstructured data (such as various documents), there are also some databases of the enterprise that need to be used as data sources in the question and answer function of AI applications. For example, users may ask the AI ​​assistant: Which new customers have been added in the past week.

### 2. Describe the feature you'd like to see

In the create database-select data source interface, in addition to importing existing text and synchronizing from notification, a new button is added: Connect to local database

![image](https://github.com/langgenius/dify/assets/31387736/2a0cd416-a882-4780-b9cf-a3320fc6105f)

### 3. How will this feature improve your workflow or experience?

After adding the enterprise's local database to the data source, the data sources of the RAG function can be greatly enriched and enterprise-level AI question and answer services can be provided.

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by LeoQuote** [comment]
Not good enough, I think it brings more frame for importing resource, how the table would be designed, what database would dify support? It would be better to stick with the api import, in addition support metadata write like document source , described in https://github.com/langgenius/dify/issues/3000

**Comment by tangxqa** [comment]
I think Dify may require the functionality of a database, even if it's just storing data in Dify instead of connecting to a local enterprise database.
Scenario:
I uploaded an Excel file containing recent news, including fields such as title, content, author, topic, time, and more. If it's a database, I can accurately find all the news under a certain topic and summarize the content through a large model
